integrations/helm/status/status.go

```diff
@@ -101,25 +85,109 @@ bail:
 	}
 
-	cFhr.Status.ReleaseName = releaseName
-	cFhr.Status.ReleaseStatus = releaseStatus
-	_, err = client.UpdateStatus(cFhr)
+	if cHr.Status.ReleaseName == releaseName && cHr.Status.ReleaseStatus == releaseStatus {
+		return nil
+	}
+
+	cHr.Status.ReleaseName = releaseName
+	cHr.Status.ReleaseStatus = releaseStatus
+
+	_, err = client.UpdateStatus(cHr)
 	return err
 }
 
-func UpdateReleaseRevision(client v1beta1client.HelmReleaseInterface, fhr v1beta1.HelmRelease, revision string) error {
-	cFhr, err := client.Get(fhr.Name, metav1.GetOptions{})
+
+// SetReleaseRevision updates the status of the HelmRelease to the
+// given revision.
+func SetReleaseRevision(client v1beta1client.HelmReleaseInterface, hr v1beta1.HelmRelease, revision string) error {
+	cHr, err := client.Get(hr.Name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
-	cFhr.Status.Revision = revision
-	_, err = client.UpdateStatus(cFhr)
+	if cHr.Status.Revision == revision {
+		return nil
+	}
+
+	cHr.Status.Revision = revision
+
+	_, err = client.UpdateStatus(cHr)
 	return err
 }
+
+// SetValuesChecksum updates the values checksum of the HelmRelease to
+// the given checksum.
+func SetValuesChecksum(client v1beta1client.HelmReleaseInterface, hr v1beta1.HelmRelease, valuesChecksum string) error {
+	cHr, err := client.Get(hr.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if valuesChecksum == "" || cHr.Status.ValuesChecksum == valuesChecksum {
+		return nil
+	}
+
+	cHr.Status.ValuesChecksum = valuesChecksum
+
+	_, err = client.UpdateStatus(cHr)
+	return err
+}
+
+// SetObservedGeneration updates the observed generation status of the
+// HelmRelease to the given generation.
+func SetObservedGeneration(client v1beta1client.HelmReleaseInterface, hr v1beta1.HelmRelease, generation int64) error {
+	cHr, err := client.Get(hr.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if cHr.Status.ObservedGeneration >= generation {
+		return nil
+	}
+
+	cHr.Status.ObservedGeneration = generation
+
+	_, err = client.UpdateStatus(cHr)
+	return err
+}
+
+// ReleaseFailed returns if the roll-out of the HelmRelease failed.
+func ReleaseFailed(hr v1beta1.HelmRelease) bool {
+	return hr.Status.ReleaseStatus == helmrelease.Status_FAILED.String()
+}
+
+
+// HasSynced returns if the HelmRelease has been processed by the
+// controller.
+func HasSynced(hr v1beta1.HelmRelease) bool {
+	return hr.Status.ObservedGeneration >= hr.Generation
+}
+
+// HasRolledBack returns if the current generation of the HelmRelease
+// has been rolled back.
+func HasRolledBack(hr v1beta1.HelmRelease) bool {
+	if !HasSynced(hr) {
+		return false
+	}
+
+	rolledBack := GetCondition(hr.Status, v1beta1.HelmReleaseRolledBack)
+	if rolledBack == nil {
+		return false
+	}
+
+	chartFetched := GetCondition(hr.Status, v1beta1.HelmReleaseChartFetched)
+	if chartFetched != nil {
+		return !(chartFetched.Status == v1.ConditionTrue && rolledBack.LastUpdateTime.Before(&chartFetched.LastUpdateTime))
```

Comment:
Can you explain (in a comment) why `LastUpdateTime` is necessary here, rather than `LastTransitionTime`?